wallet-test/src/test/scala/org/bitcoins/wallet/models/SpendingInfoDAOTest.scala

```diff
@@ -17,17 +23,110 @@ class SpendingInfoDAOTest extends BitcoinSWalletTest with WalletDAOFixture {
     val utxoDAO = daos.utxoDAO
 
     for {
-      created <- utxoDAO.create(WalletTestUtil.sampleSegwitUTXO)
+      created <- WalletTestUtil.insertSegWitUTXO(daos)
       read <- utxoDAO.read(created.id.get)
-    } yield assert(read.contains(created))
+    } yield
+      read match {
+        case None                          => fail(s"Did not read back a UTXO")
+        case Some(_: SegwitV0SpendingInfo) => succeed
+        case Some(other)                   => fail(s"did not get segwit UTXO: $other")
+      }
   }
 
   it should "insert a legacy UTXO and read it" in { daos =>
     val utxoDAO = daos.utxoDAO
     for {
-      created <- utxoDAO.create(WalletTestUtil.sampleLegacyUTXO)
+      created <- WalletTestUtil.insertLegacyUTXO(daos)
       read <- utxoDAO.read(created.id.get)
-    } yield assert(read.contains(created))
+    } yield
+      read match {
+        case None                        => fail(s"Did not read back a UTXO")
+        case Some(_: LegacySpendingInfo) => succeed
+        case Some(other)                 => fail(s"did not get a legacy UTXO: $other")
+      }
+  }
+
+  it should "find incoming outputs being spent, given a TX" in { daos =>
+    val WalletDAOs(_, _, utxoDAO) = daos
+
+    for {
+      utxo <- WalletTestUtil.insertLegacyUTXO(daos)
+      transaction = {
+        val randomTX = TransactionGenerators.transaction
+          .suchThat(_.inputs.nonEmpty)
+          .sampleSome
+
+        val inputs = {
+          val head = randomTX.inputs.head
+          val ourInput =
+            TransactionInput(utxo.outPoint,
+                             ScriptSignature.empty,
+                             head.sequence)
+
+          ourInput +: randomTX.inputs.tail
+        }
+
+        BaseTransaction(randomTX.version,
```

Comment:
nit: I don't think this matters in this case, but remember when you are going from a `Transaction` -- which can be a `WitnessTransaction`  -- to a `BaseTransaction` you may be discarding witness data.